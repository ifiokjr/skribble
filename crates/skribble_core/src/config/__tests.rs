use rstest::rstest;
use similar_asserts::assert_eq;

use super::*;
use crate::AnyEmptyResult;

#[rstest]
#[case("hsl", "hsl(314 100% 47.0588%)", "314 100% 47.0588%")]
#[case("hsl", "hsl(314 100% 47.0588% / 1.0)", "314 100% 47.0588%")]
fn color_format_inner_color(
  #[case] format: &str,
  #[case] input: &str,
  #[case] expected: &str,
) -> AnyEmptyResult {
  let color_format = ColorFormat::from(format);
  assert_eq!(color_format.get_inner_color(input)?, expected);
  Ok(())
}

#[test]
fn check_config_can_serialize() {
  let config: StyleConfig = Default::default();
  let json = config.to_json().unwrap();
  StyleConfig::from_json(json).unwrap();
}

#[test]
fn default_config() {
  insta::assert_json_snapshot!(StyleConfig::default());
}
