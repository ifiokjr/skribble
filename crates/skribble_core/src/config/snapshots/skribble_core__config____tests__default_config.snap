---
source: crates/skribble_core/src/config/__tests.rs
expression: "StyleConfig::default()"
---
{
  "options": {
    "output": "skribble.css",
    "root": "./",
    "files": [
      "**"
    ],
    "charset": "utf-8",
    "defaultLayer": "default",
    "colorFormat": "hsl",
    "mergeRules": {
      "keyframes": "append",
      "variables": "append",
      "mediaQueries": "append",
      "modifiers": "append",
      "rules": "append",
      "classes": "append",
      "palette": "append",
      "atoms": "append",
      "groups": "append"
    },
    "useRegisteredProperties": false,
    "variablePrefix": "sk",
    "defaultColor": "#000000",
    "minify": false,
    "disableFormatting": false,
    "formatters": []
  },
  "layers": [
    {
      "priority": 250,
      "value": "base"
    },
    {
      "priority": 250,
      "value": "alias"
    },
    {
      "priority": 150,
      "value": "default"
    }
  ],
  "cssChunks": [],
  "keyframes": [],
  "variables": [],
  "mediaQueries": [],
  "modifiers": [],
  "transformers": [],
  "atoms": [],
  "classes": [],
  "aliases": [],
  "palette": {},
  "valueSets": []
}
